src/test/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgentTest.java

```diff
@@ -0,0 +1,106 @@
+import com.aws.greengrass.authorization.Permission;
+import com.aws.greengrass.authorization.exceptions.AuthorizationException;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService;
+import software.amazon.awssdk.aws.greengrass.model.Metric;
+import software.amazon.awssdk.aws.greengrass.model.PutComponentMetricRequest;
+import software.amazon.awssdk.aws.greengrass.model.PutComponentMetricResponse;
+import software.amazon.awssdk.crt.eventstream.ServerConnectionContinuation;
+import software.amazon.awssdk.eventstreamrpc.AuthenticationData;
+import software.amazon.awssdk.eventstreamrpc.OperationContinuationHandlerContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.aws.greengrass.ipc.modules.ComponentMetricIPCService.PUT_COMPONENT_METRIC_SERVICE_NAME;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class ComponentMetricIPCEventStreamAgentTest {
+    private static final String TEST_COMPONENT = "aws.greengrass.testcomponent";
+
+
+    @Mock
+    OperationContinuationHandlerContext mockContext;
+    @Mock
+    AuthenticationData mockAuthenticationData;
+    @Mock
+    AuthorizationHandler authorizationHandler;
+    @Captor
+    ArgumentCaptor<Permission> permissionArgumentCaptor;
+
+    final ExecutorService pool = Executors.newCachedThreadPool();
+    private ComponentMetricIPCEventStreamAgent componentMetricIPCEventStreamAgent;
+
+    @BeforeEach
+    public void setup() {
+        lenient().when(mockContext.getContinuation()).thenReturn(mock(ServerConnectionContinuation.class));
+        lenient().when(mockContext.getAuthenticationData()).thenReturn(mockAuthenticationData);
+        lenient().when(mockAuthenticationData.getIdentityLabel()).thenReturn(TEST_COMPONENT);
+        componentMetricIPCEventStreamAgent = new ComponentMetricIPCEventStreamAgent(authorizationHandler);
+    }
+
+    @AfterEach
+    void afterEach() {
+        pool.shutdownNow();
+    }
+
+
+    @Test
+    void GIVEN_put_component_metric_request_WHEN_handle_request_called_THEN_component_passes_authorization()
```

Comment:
Is only one test enough here? We should add more tests for other authorization cases and validation, emitmetrics etc